stdlib/http/src/main/java/org/ballerinalang/net/http/WebSocketFailoverClientListener.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright (c) 2019, WSO2 Inc. (http:www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http:www.apache.orglicensesLICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.ballerinalang.net.http;
+
+import org.ballerinalang.jvm.values.ObjectValue;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.wso2.transport.http.netty.contract.websocket.WebSocketCloseMessage;
+import org.wso2.transport.http.netty.contract.websocket.WebSocketConnection;
+
+import java.io.IOException;
+
+import static org.ballerinalang.net.http.WebSocketConstants.STATUS_CODE_ABNORMAL_CLOSURE;
+import static org.ballerinalang.net.http.WebSocketUtil.determineAction;
+import static org.ballerinalang.net.http.WebSocketUtil.handleExceptionAndDispatchCloseMessage;
+import static org.ballerinalang.net.http.WebSocketUtil.hasRetryConfig;
+
+/**
+ * Failover client listener for WebSocket.
+ *
+ * @since 1.0.0
+ */
+public class WebSocketFailoverClientListener extends WebSocketClientConnectorListener {
+
+    private WebSocketOpenConnectionInfo connectionInfo;
+    private static final Logger logger = LoggerFactory.getLogger(WebSocketFailoverClientListener.class);
+
+    public void setConnectionInfo(WebSocketOpenConnectionInfo connectionInfo) {
+        this.connectionInfo = connectionInfo;
+    }
+
+    @Override
+    public void onMessage(WebSocketCloseMessage webSocketCloseMessage) {
+        ObjectValue webSocketClient = connectionInfo.getWebSocketEndpoint();
+        int statusCode = webSocketCloseMessage.getCloseCode();
+        if (statusCode == STATUS_CODE_ABNORMAL_CLOSURE) {
+            determineAction(connectionInfo, null, webSocketCloseMessage);
+        } else {
+            if (hasRetryConfig(webSocketClient)) {
+                logger.info("Couldn't connect to the server because the connected server " +
+                        "has sent the close request to the client.");
+            }
+            handleExceptionAndDispatchCloseMessage(connectionInfo, webSocketCloseMessage);
+        }
+    }
+
+    @Override
+    public void onError(WebSocketConnection webSocketConnection, Throwable throwable) {
+        if (throwable instanceof IOException) {
+            determineAction(connectionInfo, throwable, null);
+        } else {
+            logger.info("Unable do the retry because the connection has some issue that needs to fix.");
```

Comment:
I don't think we need such a message. Simply logging the error would suffice:
```suggestion
            logger.error("Error occured: ", throwable);
```